runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -6674,13 +6722,86 @@ void TR_EscapeAnalysis::makeContiguousLocalAllocation(Candidate *candidate)
       }
 
    if (candidate->escapesInColdBlocks())
+      {
       candidate->_originalAllocationNode = candidate->_node->duplicateTree();
+      }
 
    bool skipZeroInit = false;
-   if ((candidate->_node->getOpCodeValue() != TR::New) &&
+   TR::ILOpCodes       originalAllocationOpCode = candidate->_node->getOpCodeValue();
+
+   if ((originalAllocationOpCode != TR::New) && (originalAllocationOpCode != TR::newvalue) &&
        candidate->_node->canSkipZeroInitialization())
       skipZeroInit = true;
 
+   if (candidate->isExplicitlyInitialized() && (originalAllocationOpCode == TR::newvalue))
+      {
+      // TODO:  Can this code be commoned up with code in lowerNewValue?
+      //
+      TR_OpaqueClassBlock *valueClass = static_cast<TR_OpaqueClassBlock *>(candidate->_node->getFirstChild()->getSymbol()->getStaticSymbol()->getStaticAddress());
+      const TR::TypeLayout* typeLayout = comp()->typeLayout(valueClass);
+
+      TR::TreeTop *fieldValueTreeTopCursor = candidate->_treeTop->getPrevTreeTop();
+      TR::TreeTop *fieldStoreTreeTopCursor = candidate->_treeTop;
```

Comment:
Could an example of IL trees be added to help others understand what stack allocated `newvalue` trees look like compared to the origin `newvalue` IL tree?

Reply:
Fixed in commit 7ebca667516f0962fcb3fd7ab40e5501c1885389